### DISCUSSION TITLE
One click FB video call

### DISCUSSION THREAD

**Discussion by leovoon** [johnlindquist/kit#848] [Q&A] [answered]
Hi, I'm a newbie to scripting. I tried this but it did not work. Can you point me in the correct direction?

```js
const url = "https://www.facebook.com/messages/t/<conversation id>"

const js = `
document.querySelector('[aria-label="Start a video call"]').click();
`

await applescript(
  `
    tell application "Google Chrome"
        open location "${url}"
    end tell
`
)

await applescript(`
  tell application "Google Chrome" to tell window 1
      set str to execute active tab javascript "${js}"
  end tell
`)
```

**Comment by johnlindquist** [comment, marked as answer]
Hi @leovoon !

## Enabling Apple Events

First and foremost, make sure you have "Allow JavaScript from Apple Events" enabled:
![CleanShot 2022-07-22 at 10 29 09](https://user-images.githubusercontent.com/36073/180483377-40700e65-d199-4529-ba3f-ec75535c5a10.png)

## Escaping Strings for AppleScript Strings

After that, I believe it's a matter of escaping the double quotes in your `js` since the `js` is a string injecting into another string:

```js
// Notice the double \\ since we want to keep the \ when it's injected into the string
const js = `
document.querySelector('[aria-label=\\"Start a video call\\"]').click();
`
```

Since quotes aren't required for selectors, another approach would be using the `$=` to find a selector that "ends with" `call`:

```js
const js = `
document.querySelector('[aria-label$=call]').click();
`
```

## Browse

The `browse` helper will open your default browser with a URL:

```js
await browse(url)

// wait ~2 seconds for page to load before sending the click
await wait(2000)
```

Please let me know if you have any more questions. Always happy to help!

**Reply by leovoon** [reply]
Aha! Thanks for responding so quickly. Really appreciate the different approach you provided! 👍🏼

**Reply by leovoon** [reply]
Since I have few different profiles, so here's what I ended up with.

```js
exec(`open -na 'Google Chrome' ${url} --args --profile-directory="Default"`)

await applescript(`
  delay 2
  tell application "Google Chrome" to tell window 1
      set str to execute active tab javascript "${js}"
  end tell
`)
···

**Reply by johnlindquist** [reply]
Awesome, looks great!

Note: We technically don't need "set str to" since the script never uses the "str" variable. Removing it won't change anything, just a technicality.
